pkg/apis/application/v1alpha1/repository_types.go

```diff
@@ -265,6 +266,13 @@ func (m *Repository) CopySettingsFrom(source *Repository) {
 	}
 }
 
+func (m *Repository) StringForLogging() string {
+	if m == nil {
+		return ""
+	}
+	return fmt.Sprintf("&Repository{Repo: %q, Type: %q, Name: %q, Project: %q}", m.Repo, m.Type, m.Name, m.Project)
+}
```

Comment:
Hm, we could make this a true implementation of the `Stringer` interface instead?

Reply:
`.String()` is already implemented by the protobuf-generated code for the `Repository` struct. That's the method which is producing the unsanitized string.

Reply:
Yeah, that makes sense. Thanks!